lib/client/sse_client.dart

```diff
@@ -133,12 +140,48 @@ class SseClient extends StreamChannelMixin<String?> {
         _logger.warning('Invalid argument: $e');
       }
       try {
-        await HttpRequest.request('$_serverUrl&messageId=${++_lastMessageId}',
-            method: 'POST', sendData: encodedMessage, withCredentials: true);
-      } catch (e) {
-        _logger.severe('Failed to send $message:\n $e');
-        close();
+        final url = '$_serverUrl&messageId=${++_lastMessageId}';
+        await _fetch(
+            url,
+            FetchOptions(
+                method: 'POST',
+                body: encodedMessage,
+                credentialsOptions:
+                    CredentialsOptions(credentials: 'include')));
+      } catch (error) {
+        final augmentedError = 'SSE client failed to send $message:\n $error';
+        _logger.severe(augmentedError);
+        _closeWithError(augmentedError);
       }
     });
   }
 }
+
+// Custom implementation of Fetch API until Dart supports GET vs. POST,
+// credentials, etc. See https://github.com/dart-lang/http/issues/595.
+Future<dynamic> _fetch(String resourceUrl, FetchOptions options) {
+  return js_util.promiseToFuture(js_util.callMethod(globalThis, 'fetch', [
+    resourceUrl,
+    options,
+  ]));
+}
+
+@JS()
+@anonymous
+class FetchOptions {
+  external String get method; // e.g., 'GET', 'POST'
```

Comment:
[nit] I don't think these getters are used in Dart. It shouldn't be necessary to define them if they aren't used - the constructor is the important part of this class definition

```dart
class FetchOptions {
  external factory FetchOptions({
    String method, // e.g., 'GET', 'POST'
    CredentialsOptions credentialsOptions,
    String? body,
  });
}
```

Reply:
Good point, removed the getters